src/app/fdctl/run/tiles/fd_repair.c

```diff
@@ -0,0 +1,675 @@
+#include "../../../../disco/tvu/util.h"
+
+#include <unistd.h>
+#include <arpa/inet.h>
+#include <linux/unistd.h>
+#include <sys/random.h>
+#include <netdb.h>
+#include <netinet/in.h>
+#include <sys/socket.h>
+
+
+#include "../../../../util/net/fd_eth.h"
+#include "../../../../util/net/fd_ip4.h"
+#include "../../../../util/net/fd_udp.h"
+
+#define NET_IN_IDX      0
+#define CONTACT_IN_IDX  1
+#define STAKE_IN_IDX    2
+#define STORE_IN_IDX    3
+
+#define STORE_OUT_IDX   0
+
+#define MAX_REPAIR_PEERS 40200UL
+
+#define SMAX    (1UL << 30UL)
+#define SDEPTH  (1UL << 11UL)
+
+
+static int
+fd_pubkey_eq( fd_pubkey_t const * key1, fd_pubkey_t const * key2 ) {
+  return memcmp( key1->key, key2->key, sizeof(fd_pubkey_t) ) == 0;
+}
+
+static ulong
+fd_pubkey_hash( fd_pubkey_t const * key, ulong seed ) {
+  return fd_hash( seed, key->key, sizeof(fd_pubkey_t) ); 
+}
+
+static void
+fd_pubkey_copy( fd_pubkey_t * keyd, fd_pubkey_t const * keys ) {
+  memcpy( keyd->key, keys->key, sizeof(fd_pubkey_t) );
+}
+
+/* Contact info table */
+#define MAP_NAME     fd_contact_info_table
+#define MAP_KEY_T    fd_pubkey_t
+#define MAP_KEY_EQ   fd_pubkey_eq
+#define MAP_KEY_HASH fd_pubkey_hash
+#define MAP_KEY_COPY fd_pubkey_copy
+#define MAP_T        fd_contact_info_elem_t
+#include "../../../../util/tmpl/fd_map_giant.c"
+
+
+struct fd_repair_tile_ctx {
+  fd_repair_t * repair;
+  fd_repair_config_t repair_config;
+
+  fd_repair_peer_addr_t repair_my_intake_addr;
+  fd_repair_peer_addr_t repair_my_serve_addr;
+  ushort                repair_intake_listen_port;
+  ushort                repair_serve_listen_port;
+
+  uchar       identity_private_key[ 32 ];
+  fd_pubkey_t identity_public_key;
+
+  fd_wksp_t * wksp;
+
+  fd_wksp_t * contact_in_mem;
+  ulong       contact_in_chunk0;
+  ulong       contact_in_wmark;
+
+  fd_wksp_t * stake_weights_in_mem;
+  ulong       stake_weights_in_chunk0;
+  ulong       stake_weights_in_wmark;
+
+  fd_wksp_t * repair_req_in_mem;
+  ulong       repair_req_in_chunk0;
+  ulong       repair_req_in_wmark;
+
+  fd_wksp_t *     net_in;
+  ulong           chunk;
+  ulong           wmark;
+
+  fd_frag_meta_t * net_out_mcache;
+  ulong *          net_out_sync;
+  ulong            net_out_depth;
+  ulong            net_out_seq;
+
+  fd_wksp_t * net_out_mem;
+  ulong       net_out_chunk0;
+  ulong       net_out_wmark;
+  ulong       net_out_chunk;
+
+  fd_frag_meta_t * store_out_mcache;
+  ulong *          store_out_sync;
+  ulong            store_out_depth;
+  ulong            store_out_seq;
+
+  fd_wksp_t * store_out_mem;
+  ulong       store_out_chunk0;
+  ulong       store_out_wmark;
+  ulong       store_out_chunk;
+
+  uchar src_mac_addr[6];
+
+  /* Includes Ethernet, IP, UDP headers */
+  ulong repair_buffer_sz;
+  uchar repair_buffer[ FD_NET_MTU ];
+};
+typedef struct fd_repair_tile_ctx fd_repair_tile_ctx_t;
+
+static fd_gossip_peer_addr_t *
+resolve_hostport( const char * str /* host:port */, fd_gossip_peer_addr_t * res ) {
```

Comment:
This function is problematic in a few ways and could probably just be deleted outright

 * The hand rolled string parsing is... suspicious at best, and doesn't appear necessary either.
 * You can't call gethostname here, you're inside the sandbox. Gethostname is also not going to work well for what you want to do. I'd suggest just starting with an ip address field, and defaulting it to what gets determined by the net tile anyway.
 * It's not in a firedancer code style, and eg htons can be fd_ushort_bswap, bad indentation etc

Reply:
I might need some clarification from @asiegel-jt on the reasoning behind this setup. It would also be nice to get a quick explanation on what "gets determined by the net tile" vs gethostname means in reality (or with an example).

Reply:
this was a convenient syntax for command line parsing. hard coding ip addresses in configs is fragile and error prone.

Reply:
this function is already gone on my integration branch. wait up for that where we really clean this up.